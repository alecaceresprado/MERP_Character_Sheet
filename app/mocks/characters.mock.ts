import { RacesEnum, ObjectTypesEnum } from 'enums';
import { classes } from '../constants';
import { Player } from 'models/player.model';

export const characters: Player[] = [
  {
    id: 1,
    details: {
      name: 'Lothar Picco',
      nickName: 'Loti Picco',
      race: RacesEnum.NORSEMEN,
      height: '2 meters',
      weight: '110KG',
      hairColor: 'Red',
      eyesColor: 'Blue',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.WARRIOR.classId,
      experience: 31751,
      level: 3
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: [
      {
        name: 'regular chain mail',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor', 'chainMail']],
        modifier: 0
      },
      {
        name: 'enhanced arm guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['attack']],
        modifier: 0
      },
      {
        name: 'enhanced leg guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor']],
        modifier: 0
      },
      {
        name: 'throwing knives',
        quantity: 5,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'throw']],
        modifier: 0,
        description: 'personally crafted, shaped like raven feathers'
      },
      {
        name: 'winter vigil - magic axe',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [
          ['weapons', 'dualWeapons'],
          ['weapons', 'projectile']
        ],
        modifier: 10,
        description:
          'one handed axe: magic ability to eject axe`s head and then summon it back'
      },
      {
        name: 'regular sword',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'dualWeapons']],
        modifier: 0,
        description: 'one handed sword'
      },
      {
        name: 'special knives belt',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables to grab and throw on the same round'
      },
      {
        name: 'enhanced body cover (embozo)',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['survival', 'hide']],
        modifier: 10
      },
      {
        name: 'Charms neck lace',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'Raven charm'
      },
      {
        name: 'magic anklet (tobillera)',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables the user to cast the "world traveler" magic'
      },
      {
        name: 'odyn',
        quantity: 1,
        type: ObjectTypesEnum.PET,
        description: 'Raven, bound to me on my trials'
      },
      {
        name: 'gold coins',
        quantity: 1,
        type: ObjectTypesEnum.CURRENCY,
        description: ''
      }
    ]
  },
  {
    id: 2,
    details: {
      name: 'Legolas Greenleaf',
      nickName: 'Lembas',
      race: RacesEnum.SINDAR_ELF,
      height: '1,82 meters',
      weight: '25KG',
      hairColor: 'Blonde',
      eyesColor: 'Blue',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.RANGER.classId,
      experience: 0,
      level: 1
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: []
  },
  {
    id: 3,
    details: {
      name: 'Boromir son of Denethor',
      nickName: 'Boromir',
      race: RacesEnum.MAN,
      height: '1.93 meters',
      weight: '85KG',
      hairColor: 'Dark',
      eyesColor: 'Grey',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.WARRIOR.classId,
      experience: 0,
      level: 0
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: [
      {
        name: 'regular chain mail',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor', 'chainMail']],
        modifier: 0
      },
      {
        name: 'enhanced arm guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['attack']],
        modifier: 0
      },
      {
        name: 'enhanced leg guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor']],
        modifier: 0
      },
      {
        name: 'throwing knives',
        quantity: 5,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'throw']],
        modifier: 0,
        description: 'personally crafted, shaped like raven feathers'
      },
      {
        name: 'winter vigil - magic axe',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [
          ['weapons', 'dualWeapons'],
          ['weapons', 'projectile']
        ],
        modifier: 10,
        description:
          'one handed axe: magic ability to eject axe`s head and then summon it back'
      },
      {
        name: 'regular sword',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'dualWeapons']],
        modifier: 0,
        description: 'one handed sword'
      },
      {
        name: 'special knives belt',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables to grab and throw on the same round'
      },
      {
        name: 'enhanced body cover (embozo)',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['survival', 'hide']],
        modifier: 10
      },
      {
        name: 'Charms neck lace',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'Raven charm'
      },
      {
        name: 'magic anklet (tobillera)',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables the user to cast the "world traveler" magic'
      },
      {
        name: 'odyn',
        quantity: 1,
        type: ObjectTypesEnum.PET,
        description: 'Raven, bound to me on my trials'
      },
      {
        name: 'gold coins',
        quantity: 1,
        type: ObjectTypesEnum.CURRENCY,
        description: ''
      }
    ]
  },
  {
    id: 4,
    details: {
      name: 'Aragorn II',
      nickName: 'Estel',
      race: RacesEnum.DUNEDAIN,
      height: '1.85 meters',
      weight: '80KG',
      hairColor: 'Dark',
      eyesColor: 'Grey',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.RANGER.classId,
      experience: 0,
      level: 0
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: [
      {
        name: 'regular chain mail',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor', 'chainMail']],
        modifier: 0
      },
      {
        name: 'enhanced arm guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['attack']],
        modifier: 0
      },
      {
        name: 'enhanced leg guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor']],
        modifier: 0
      },
      {
        name: 'throwing knives',
        quantity: 5,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'throw']],
        modifier: 0,
        description: 'personally crafted, shaped like raven feathers'
      },
      {
        name: 'winter vigil - magic axe',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [
          ['weapons', 'dualWeapons'],
          ['weapons', 'projectile']
        ],
        modifier: 10,
        description:
          'one handed axe: magic ability to eject axe`s head and then summon it back'
      },
      {
        name: 'regular sword',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'dualWeapons']],
        modifier: 0,
        description: 'one handed sword'
      },
      {
        name: 'special knives belt',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables to grab and throw on the same round'
      },
      {
        name: 'enhanced body cover (embozo)',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['survival', 'hide']],
        modifier: 10
      },
      {
        name: 'Charms neck lace',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'Raven charm'
      },
      {
        name: 'magic anklet (tobillera)',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables the user to cast the "world traveler" magic'
      },
      {
        name: 'odyn',
        quantity: 1,
        type: ObjectTypesEnum.PET,
        description: 'Raven, bound to me on my trials'
      },
      {
        name: 'gold coins',
        quantity: 1,
        type: ObjectTypesEnum.CURRENCY,
        description: ''
      }
    ]
  }
];
