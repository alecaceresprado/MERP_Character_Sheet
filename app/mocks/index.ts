export * from './characters.mock';
export * from './list.mock';
