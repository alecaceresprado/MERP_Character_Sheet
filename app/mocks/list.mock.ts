import { PlayerSummary } from 'models/player-summary.model';

export const playerList: PlayerSummary[] = [
  {
    id: '1',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  },
  {
    id: '2',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  },
  {
    id: '3',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  },
  {
    id: '4',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  }
];
