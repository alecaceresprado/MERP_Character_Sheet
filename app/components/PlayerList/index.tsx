import * as React from 'react';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';

import LoadingIndicator from 'components/LoadingIndicator';
import { AppState } from 'state/app/app.reducer';
import PlayerSummaryCard from 'containers/PlayerSummaryCard';

export type PlayerListProps = Pick<AppState, 'loading' | 'error' | 'players'>;

const useStyles = makeStyles(theme => ({
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8)
  }
}));

function PlayerList({ loading, error, players }: PlayerListProps) {
  const classes = useStyles();

  if (loading) {
    return <LoadingIndicator />;
  }

  if (error !== false || players === undefined) {
    return <div>Something went wrong, please try again!</div>;
  }

  if (!players.length) {
    return null;
  }

  return (
    <Container className={classes.cardGrid} maxWidth="md">
      <Grid container spacing={4}>
        {players.map(player => (
          <PlayerSummaryCard key={`item-${player.id}`} item={player} />
        ))}
      </Grid>
    </Container>
  );
}

export default PlayerList;
