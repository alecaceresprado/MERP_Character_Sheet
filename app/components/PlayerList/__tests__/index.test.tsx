import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import PlayerList from '../index';
import configureStore from '../../../configureStore';
import history from '../../../utils/history';
import { playerList } from 'mocks';

describe('<PlayerList />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<PlayerList players={[]} loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <PlayerList
          players={[]}
          loading={false}
          error={{ message: 'Loading failed!' }}
        />
      </IntlProvider>
    );
    expect(queryByText(/Something went wrong/)).toBeInTheDocument();
  });

  it('should render the repositories if loading was successful', () => {
    const initialState = {
      global: {
        error: false,
        loading: false,
        players: []
      }
    };
    const store = configureStore(initialState, history);
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <PlayerList players={playerList} error={false} loading={false} />
        </IntlProvider>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing interesting is provided', () => {
    const { container } = render(
      <PlayerList players={[]} error={false} loading={false} />
    );

    expect(container).toBeEmpty();
  });
});
