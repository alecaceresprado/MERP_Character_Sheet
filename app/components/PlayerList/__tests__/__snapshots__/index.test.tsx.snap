// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerList /> should render the loading indicator when its loading 1`] = `
<div
  class="sc-bdVaJa kAPAHy"
>
  <div
    class="sc-bwzfXH kHYkBk"
  />
  <div
    class="sc-htpNat jSPapt"
  />
  <div
    class="sc-bxivhb leTqfi"
  />
  <div
    class="sc-ifAKCX lkOVGt"
  />
  <div
    class="sc-EHOje chnvPL"
  />
  <div
    class="sc-bZQynM iqYbZJ"
  />
  <div
    class="sc-gzVnrw CCqTk"
  />
  <div
    class="sc-htoDjs bWxymj"
  />
  <div
    class="sc-dnqmqq ieSRwc"
  />
  <div
    class="sc-iwsKbI cdedUB"
  />
  <div
    class="sc-gZMcBi bhuLmD"
  />
  <div
    class="sc-gqjmRU kpUady"
  />
</div>
`;

exports[`<PlayerList /> should render the repositories if loading was successful 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthMd"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardMedia-root"
          style="background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU);"
          title="Image title"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Lothar Picco
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Braided long beard with charms with runes
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/1"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                View
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/1/edit"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                Edit
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardMedia-root"
          style="background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU);"
          title="Image title"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Lothar Picco
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Braided long beard with charms with runes
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/2"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                View
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/2/edit"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                Edit
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardMedia-root"
          style="background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU);"
          title="Image title"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Lothar Picco
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Braided long beard with charms with runes
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/3"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                View
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/3/edit"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                Edit
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardMedia-root"
          style="background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU);"
          title="Image title"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Lothar Picco
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Braided long beard with charms with runes
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/4"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                View
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/player/4/edit"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span>
                Edit
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
